client/src/components/App/Content/InvestigationForm/InvestigationForm.tsx

```diff
@@ -65,19 +65,21 @@ const InvestigationForm: React.FC = (): JSX.Element => {
             setSymptomsStartDate, setExposureDate, setHasSymptoms, setEndInvestigationDate]
     );
 
-    const { currentTab, setCurrentTab, confirmFinishInvestigation, handleSwitchTab } = useInvestigationForm({ clinicalDetailsVariables, personalInfoData });
+    const { currentTab, setCurrentTab, confirmFinishInvestigation, handleSwitchTab, saveCurrentTab } = useInvestigationForm({ clinicalDetailsVariables, personalInfoData });
 
     return (
         <div className={classes.content}>
             <PersonalInfoContextProvider value={personalInfoValue}>
                 <ClinicalDetailsDataContextProvider value={clinicalDetailsVariables}>
                     <StartInvestigationDateVariablesProvider value={startInvestigationDateVariables}>
-                        <InvestigationInfoBar />
+                        <InvestigationInfoBar
+                            onExitInvestigation={saveCurrentTab}
+                        />
                         <div className={classes.interactiveForm}>
                             <TabManagement
                                 currentTab={currentTab}
                                 setCurrentTab={setCurrentTab}
-                                onTabClicked={handleSwitchTab}
+                                onTabClicked={saveCurrentTab}
```

Comment:
Where are you dealing the tab switch?

Reply:
If I click on the tab then I need to save the data in the current tab, in the meantime it is not disabled

Reply:
I mean cant you move to the next tab when clicking the tab?

Reply:
No I need to save the tab data before that

Reply:
Yes, but where is the tab switch itself?